.github/workflows/misc.yml

```diff
@@ -62,4 +67,27 @@ jobs:
         grep -r -n -I --color=always --exclude-dir=.git -P "$PATTERN" . || exit 0
         echo "::error::Remove zero-width/format, separator or combining-mark characters"
         exit 1
+        
+    - name: Check configs JSON format
+      run: |
+        files=("config_example.json" "testdata/configtest.json")
+        for file in "${files[@]}"; do
+          processed_file="${file%.*}_processed.${file##*.}"
+          jq '.exchanges |= sort_by(.name)' --indent 1 $file > $processed_file
+          if ! diff $file $processed_file; then
+            echo "jq differences found in $file! Please run 'make lint_configs'"
+            exit 1
+          else
+            rm $processed_file
+            echo "No differences found in $file 🌞"
+          fi
+        done
+    
+    - name: Check config versions are continuous
+      run: go test ./config/versions/ -tags config_versions -run Continuity
```

Comment:
The idea behind this being in a separate workflow is that it's Okay for it to fail right up until a PR is merged.
So what I wanted with it being separate was that we can quickly see to ignore that entire workflow on PRs during review.

If we move it into misc.yml then everyone has to open up the CI for that workflow to check if it was that or not.

I should have left it as a comment inside the yaml file.

So, I think my justification stands and config version continuity should remain a separate and easy to ignore workflow, but with a new comment explaining why not to fold it into misc.

Reply:
Yep I understand and remember your motivation; similar to how I've left the spellchecker and proto checks alone. The main caveat is how infrequently we do config upgrades, if misc fails for whatever reason they'd have to check it once every blue moon

Reply:
Okay. I've considered that.
I stand by it, though, because for a given PR that does introduce a config upgrade, they'd be checking that same thing every time they garden their PR.
🙏